### ISSUE TITLE
Startup fails with "Cannot achieve consistency level for cl ALL. Requires 3, alive 2" error after Scylla K8S pod recreation

### ISSUE THREAD

**Issue by vponomaryov** [scylladb/scylladb#14440]
## Issue description

- [ ] This issue is a regression.
- [x] It is unknown if this issue is a regression.

The test run was running in the K8S environment.

The scenario applied in the failed test is following:
- Drain K8S node that hosts Scylla pod (`kubectl drain gke-longev-scylla-operat--scylla-pool-ee517bc6-x5mp -n scylla --ignore-daemonsets --delete-local-data`)
- Wait for pod eviction
- Uncordon node
- Decrease number of ScyllaCluster members from `3` to `2`.
- Wait for the pod deletion (`kubectl --namespace=scylla wait --timeout=21600s --for=delete pod sct-cluster-us-east1-b-us-east1-2`)
- Increase number of ScyllaCluster members from `2` to `3`
- Wait for the new pod to become up and running

It is ~equivalent of the Scylla VM reboot.

So, after the creation of the third pod we see following:

```
I0625 10:52:04.829192       1 operator/sidecar.go:147] sidecar version "v1.9.0-rc.1-0-g57a2b85"
I0625 10:52:04.829277       1 flag/flags.go:64] FLAG: --burst="5"
I0625 10:52:04.829286       1 flag/flags.go:64] FLAG: --cpu-count="7"
I0625 10:52:04.829290       1 flag/flags.go:64] FLAG: --feature-gates="AllAlpha=false,AllBeta=false,AutomaticTLSCertificates=false"
I0625 10:52:04.829301       1 flag/flags.go:64] FLAG: --help="false"
I0625 10:52:04.829306       1 flag/flags.go:64] FLAG: --kubeconfig=""
I0625 10:52:04.829311       1 flag/flags.go:64] FLAG: --loglevel="2"
I0625 10:52:04.829315       1 flag/flags.go:64] FLAG: --namespace="scylla"
I0625 10:52:04.829319       1 flag/flags.go:64] FLAG: --qps="2"
I0625 10:52:04.829324       1 flag/flags.go:64] FLAG: --service-name="sct-cluster-us-east1-b-us-east1-2"
I0625 10:52:04.829327       1 flag/flags.go:64] FLAG: --v="2"
I0625 10:52:04.829724       1 operator/sidecar.go:193] "Waiting for single service informer caches to sync"
...
INFO  2023-06-25 10:52:06,756 [shard 0] iotune - Disk parameters: max_iodepth=3069 disks_per_array=3 minimum_io_size=4096
Starting Evaluation. This may take a while...
...
Measuring random read IOPS: 539838 IOPS
Writing result to /etc/scylla.d/io_properties.yaml
Writing result to /etc/scylla.d/io.conf
tuning /sys/devices/virtual/block/md127
tuning: /sys/devices/virtual/block/md127/queue/nomerges 2
warning: unable to tune /sys/devices/virtual/block/md127/queue/nomerges to 2
tuning /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n1
tuning: /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n1/queue/nomerges 2
warning: unable to tune /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n1/queue/nomerges to 2
tuning /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n2
tuning: /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n2/queue/nomerges 2
warning: unable to tune /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n2/queue/nomerges to 2
tuning /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n3
tuning: /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n3/queue/nomerges 2
warning: unable to tune /sys/devices/pci0000:00/0000:00:04.0/nvme/nvme0/nvme0n3/queue/nomerges to 2
E0625 10:54:07.383420       1 sidecar/probes.go:154] "healthz probe: can't connect to Scylla API" err="dial tcp 127.0.0.1:10000: connect: connection refused" Service="scylla/sct-cluster-us-east1-b-us-east1-2"
2023-06-25 10:54:07,499 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
2023-06-25 10:54:07,499 INFO Included extra file "/etc/supervisord.conf.d/rsyslog.conf" during parsing
...
INFO  2023-06-25 10:54:13,324 [shard 0] repair - Loading repair history for keyspace=system, table=scylla_table_schema_history, table_uuid=0191a53e-40f0-31d4-9171-b0d19ffb17b4
INFO  2023-06-25 10:54:13,324 [shard 0] init - starting CDC Generation Management service
INFO  2023-06-25 10:54:13,324 [shard 0] init - starting CDC log service
INFO  2023-06-25 10:54:13,325 [shard 0] init - starting storage service
INFO  2023-06-25 10:54:13,326 [shard 0] init - starting sstables loader
INFO  2023-06-25 10:54:13,326 [shard 0] messaging_service - Starting Messaging Service on port 7000
INFO  2023-06-25 10:54:13,327 [shard 0] storage_service - entering STARTING mode
INFO  2023-06-25 10:54:13,327 [shard 0] storage_service - Loading persisted ring state
INFO  2023-06-25 10:54:13,329 [shard 3] compaction - [Compact system.truncated 9f1b0290-1346-11ee-b39a-60c6c1f4934e] Compacted 2 sstables to [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/mc-24-big-Data.db:level=0]. 180kB to 90kB (~50% of original) in 19ms = 4MB/s. ~256 total partitions merged to 3.
INFO  2023-06-25 10:54:13,329 [shard 4] compaction - [Compact system.truncated 9f1b29a0-1346-11ee-b97a-60c9c1f4934e] Compacted 2 sstables to [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/mc-25-big-Data.db:level=0]. 81kB to 40kB (~50% of original) in 19ms = 2MB/s. ~256 total partitions merged to 1.
INFO  2023-06-25 10:54:13,330 [shard 6] compaction - [Compact system.truncated 9f1b0290-1346-11ee-80d9-60cac1f4934e] Compacted 2 sstables to [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/mc-27-big-Data.db:level=0]. 196kB to 98kB (~50% of original) in 20ms = 4MB/s. ~256 total partitions merged to 2.
INFO  2023-06-25 10:54:13,331 [shard 0] storage_service - initial_contact_nodes={10.120.4.242}, loaded_endpoints={}, loaded_peer_features=0
INFO  2023-06-25 10:54:13,331 [shard 0] storage_service - Checking remote features with gossip
INFO  2023-06-25 10:54:13,331 [shard 0] gossip - Gossip shadow round started with nodes={10.120.4.242}
INFO  2023-06-25 10:54:13,333 [shard 0] gossip - Gossip shadow round finisehd with nodes_talked={10.120.4.242}
...
INFO  2023-06-25 10:54:27,124 [shard 0] features - Feature WORKLOAD_PRIORITIZATION is enabled
INFO  2023-06-25 10:54:27,124 [shard 0] storage_service - entering JOINING mode
...
INFO  2023-06-25 10:54:27,162 [shard 0] storage_service - enable_repair_based_node_ops=true, allowed_repair_based_node_ops={replace}
INFO  2023-06-25 10:54:27,162 [shard 0] storage_service - entering BOOTSTRAP mode
INFO  2023-06-25 10:54:27,162 [shard 0] storage_service - Wait until local node knows tokens of peer nodes
INFO  2023-06-25 10:54:27,162 [shard 0] gossip - Waiting for pending range setup...
INFO  2023-06-25 10:54:27,175 [shard 2] compaction - [Compact system.local a75d6ba0-1346-11ee-b9d1-60c7c1f4934e] Compacted 2 sstables to [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/me-51-big-Data.db:level=0]. 86kB to 45kB (~52% of original) in 10ms = 4MB/s. ~256 total partitions merged to 1.
E0625 10:54:27.384448       1 sidecar/probes.go:110] "readyz probe: can't get scylla native transport" err="giving up after 1 attempts: agent [HTTP 404] Not found" Service="scylla/sct-cluster-us-east1-b-us-east1-2" Node="10.120.6.153"
E0625 10:54:37.384723       1 sidecar/probes.go:110] "readyz probe: can't get scylla native transport" err="giving up after 1 attempts: agent [HTTP 404] Not found" Service="scylla/sct-cluster-us-east1-b-us-east1-2" Node="10.120.6.153"
E0625 10:54:47.384289       1 sidecar/probes.go:110] "readyz probe: can't get scylla native transport" err="giving up after 1 attempts: agent [HTTP 404] Not found" Service="scylla/sct-cluster-us-east1-b-us-east1-2" Node="10.120.6.153"
I0625 10:54:49.624401       1 sidecar/sync.go:135] "Successfully updated HostID annotation" Service="scylla/sct-cluster-us-east1-b-us-east1-2" HostID="b1d0fcc5-eec7-42a6-b8ac-6c6aab8d74c1"
E0625 10:54:57.384669       1 sidecar/probes.go:110] "readyz probe: can't get scylla native transport" err="giving up after 1 attempts: agent [HTTP 404] Not found" Service="scylla/sct-cluster-us-east1-b-us-east1-2" Node="10.120.6.153"
INFO  2023-06-25 10:55:04,163 [shard 0] gossip - No gossip backlog; proceeding
```

But when the `cdc insert` operation step is reached:
```
INFO  2023-06-25 10:55:04,180 [shard 0] cdc - Inserting new generation data at UUID 404cf5fa-52fe-433a-9cab-5d16f2df4126
```

The scylla service starts the shut down process after the `cdc insert` operation in `4ms`:
```
INFO  2023-06-25 10:55:04,184 [shard 0] init - Shutting down group 0 service
INFO  2023-06-25 10:55:04,184 [shard 0] init - Shutting down group 0 service was successful
INFO  2023-06-25 10:55:04,184 [shard 0] init - Shutting down storage service notifications
...
INFO  2023-06-25 10:55:05,375 [shard 0] init - Shutting down sighup
INFO  2023-06-25 10:55:05,375 [shard 0] init - Shutting down sighup was successful
```

And finally the startup fails with CL error:
```
ERROR 2023-06-25 10:55:05,375 [shard 0] init - Startup failed: exceptions::unavailable_exception (Cannot achieve consistency level for cl ALL. Requires 3, alive 2)
2023-06-25 10:55:05,444 INFO exited: scylla (exit status 1; not expected)
2023-06-25 10:55:06,447 INFO spawned: 'scylla' with pid 224
Scylla version 2022.2.6-0.20230423.b9a1478dba07 with build-id d9def2e0e62ae9483d4becd80334de8b262894c4 starting ...
```

After it K8S tries to restart the failed Scylla pod and fails with another error:
```
INFO  2023-06-25 10:55:07,711 [shard 0] storage_service - Checking remote features with gossip      
INFO  2023-06-25 10:55:07,711 [shard 0] gossip - Gossip shadow round started with nodes={10.120.4.242, 10.120.2.119, 10.120.1.255}
INFO  2023-06-25 10:55:07,712 [shard 4] compaction - [Compact system.peers bf7f8d30-1346-11ee-8a7d-2eebe48c1175] Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/me-39-big-Data.db:level=0      ]. 90kB to 49kB (~54% of original) in 56ms = 877kB/s. ~256 total partitions merged to 2.
INFO  2023-06-25 10:55:07,714 [shard 3] compaction - [Compact system.peers bf83abe0-1346-11ee-bf77-2ee7e48c1175] Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/me-38-big-Data.db:level=0      ]. 86kB to 45kB (~52% of original) in 32ms = 1MB/s. ~256 total partitions merged to 1.
WARN  2023-06-25 10:55:12,711 [shard 0] gossip - The get_endpoint_states verb to node 10.120.1.255 was timeout
INFO  2023-06-25 10:55:12,711 [shard 0] gossip - Gossip shadow round finisehd with nodes_talked={10.120.2.119, 10.120.4.242}
INFO  2023-06-25 10:55:12,712 [shard 0] gossip - Feature check passed. Local node 10.120.6.153 features = {AGGREGATE_STORAGE_OPTIONS, ALTERNATOR_STREAMS, ALTERNATOR_TTL, CDC, CDC_GENERATIONS_V2, COMPUTED_COLUMNS, CORRECT_COUNTER_O      RDER, CORRECT_IDX_TOKEN_IN_SECONDARY_INDEX, CORRECT_NON_COMPOUND_RANGE_TOMBSTONES, CORRECT_STATIC_COMPACT_IN_MC, COUNTERS, DIGEST_FOR_NULL_VALUES, DIGEST_INSENSITIVE_TO_EXPIRY, DIGEST_MULTIPARTITION_READ, HINTED_HANDOFF_SEPARATE_C      ONNECTION, INDEXES, IN_MEMORY_TABLES, KEYSPACE_STORAGE_OPTIONS, LARGE_PARTITIONS, LA_SSTABLE_FORMAT, LWT, MATERIALIZED_VIEWS, MC_SSTABLE_FORMAT, MD_SSTABLE_FORMAT, ME_SSTABLE_FORMAT, NONFROZEN_UDTS, PARALLELIZED_AGGREGATION, PER_T      ABLE_CACHING, PER_TABLE_PARTITIONERS, RANGE_SCAN_DATA_VARIANT, RANGE_TOMBSTONES, ROLES, ROW_LEVEL_REPAIR, SCHEMA_COMMITLOG, SCHEMA_TABLES_V3, SEPARATE_PAGE_SIZE_AND_SAFETY_LIMIT, STREAM_WITH_RPC_STREAM, TOMBSTONE_GC_OPTIONS, TRUNC      ATION_TABLE, TYPED_ERRORS_IN_READ_RPC, UDA, UDA_NATIVE_PARALLELIZED_AGGREGATION, UNBOUNDED_RANGE_TOMBSTONES, VIEW_VIRTUAL_COLUMNS, WORKLOAD_PRIORITIZATION, WRITE_FAILURE_REPLY, XXHASH}, Remote common_features = {AGGREGATE_STORAGE_      OPTIONS, ALTERNATOR_STREAMS, ALTERNATOR_TTL, CDC, CDC_GENERATIONS_V2, COMPUTED_COLUMNS, CORRECT_COUNTER_ORDER, CORRECT_IDX_TOKEN_IN_SECONDARY_INDEX, CORRECT_NON_COMPOUND_RANGE_TOMBSTONES, CORRECT_STATIC_COMPACT_IN_MC, COUNTERS, DI      GEST_FOR_NULL_VALUES, DIGEST_INSENSITIVE_TO_EXPIRY, DIGEST_MULTIPARTITION_READ, HINTED_HANDOFF_SEPARATE_CONNECTION, INDEXES, IN_MEMORY_TABLES, KEYSPACE_STORAGE_OPTIONS, LARGE_PARTITIONS, LA_SSTABLE_FORMAT, LWT, MATERIALIZED_VIEWS,       MC_SSTABLE_FORMAT, MD_SSTABLE_FORMAT, ME_SSTABLE_FORMAT, NONFROZEN_UDTS, PARALLELIZED_AGGREGATION, PER_TABLE_CACHING, PER_TABLE_PARTITIONERS, RANGE_SCAN_DATA_VARIANT, RANGE_TOMBSTONES, ROLES, ROW_LEVEL_REPAIR, SCHEMA_COMMITLOG, S      CHEMA_TABLES_V3, SEPARATE_PAGE_SIZE_AND_SAFETY_LIMIT, STREAM_WITH_RPC_STREAM, TOMBSTONE_GC_OPTIONS, TRUNCATION_TABLE, TYPED_ERRORS_IN_READ_RPC, UDA, UDA_NATIVE_PARALLELIZED_AGGREGATION, UNBOUNDED_RANGE_TOMBSTONES, VIEW_VIRTUAL_COL      UMNS, WORKLOAD_PRIORITIZATION, WRITE_FAILURE_REPLY, XXHASH}
INFO  2023-06-25 10:55:12,712 [shard 0] init - Shutting down group 0 service 
...
ERROR 2023-06-25 10:55:12,849 [shard 0] init - Startup failed: std::runtime_error (A node with address 10.120.6.153 already exists, cancelling join. Use replace_address if you want to replace this node.)
023-06-25 10:55:12,904 INFO exited: scylla (exit status 1; not expected)                           
2023-06-25 10:55:13,906 INFO spawned: 'scylla' with pid 255                                         
Scylla version 2022.2.6-0.20230423.b9a1478dba07 with build-id d9def2e0e62ae9483d4becd80334de8b262894c4 starting ...

```

Then errors repeats and the pod never comes up again.

## Impact

Scylla member fails to start.

## How frequently does it reproduce?

First time observed.

## Installation details

Kernel Version: 5.15.0-1034-gke
Scylla version (or git commit hash): `2022.2.6-20230423.b9a1478dba07` with build-id `d9def2e0e62ae9483d4becd80334de8b262894c4`

Operator Image: scylladb/scylla-operator:1.9.0-rc.1
Operator Helm Version: 1.9.0-rc.1
Operator Helm Repository: https://storage.googleapis.com/scylla-operator-charts/latest
Cluster size: 3 nodes (n1-standard-8)

Scylla Nodes used in this run:
- sct-cluster-us-east1-b-us-east1-0 - 10.120.4.242 ; PodIP: 10.116.4.6
- sct-cluster-us-east1-b-us-east1-1 - 10.120.2.119 ; PodIP: 10.116.5.6
- sct-cluster-us-east1-b-us-east1-2 - 10.120.6.153 ; PodIP: 10.116.6.9 (Old IPs: 10.116.6.6, 10.116.6.8)

OS / Image: `scylladb/scylladb:2022.2.6`

Test: `longevity-scylla-operator-3h-gke-terminate-decommission-add`
Test id: `12145449-7c1a-44ac-aaab-8c0fc0b91299`
Test name: `scylla-operator/operator-1.9/gke/longevity-scylla-operator-3h-gke-terminate-decommission-add`
Test config file(s):

- [longevity-scylla-operator-3h-terminate-decommission-add.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/96fc31362c7d2836d8481eac6e16b6a3d267db31/test-cases/scylla-operator/longevity-scylla-operator-3h-terminate-decommission-add.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 12145449-7c1a-44ac-aaab-8c0fc0b91299`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=12145449-7c1a-44ac-aaab-8c0fc0b91299)
- Show all stored logs command: `$ hydra investigate show-logs 12145449-7c1a-44ac-aaab-8c0fc0b91299`

## Logs:
- **db-cluster-12145449.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/db-cluster-12145449.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/db-cluster-12145449.tar.gz)
- **sct-runner-events-12145449.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/sct-runner-events-12145449.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/sct-runner-events-12145449.tar.gz)
- **sct-12145449.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/sct-12145449.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/sct-12145449.log.tar.gz)
- **monitor-set-12145449.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/monitor-set-12145449.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/monitor-set-12145449.tar.gz)
- **loader-set-12145449.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/loader-set-12145449.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/loader-set-12145449.tar.gz)
- **kubernetes-12145449.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/kubernetes-12145449.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/kubernetes-12145449.tar.gz)
- **parallel-timelines-report-12145449.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/parallel-timelines-report-12145449.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/12145449-7c1a-44ac-aaab-8c0fc0b91299/20230625_114019/parallel-timelines-report-12145449.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-operator/job/operator-1.9/job/gke/job/longevity-scylla-operator-3h-gke-terminate-decommission-add/4/)
[Argus](https://argus.scylladb.com/test/8dff9295-56f7-494f-aa6a-3dcabbfacd7d/runs?additionalRuns[]=12145449-7c1a-44ac-aaab-8c0fc0b91299)
</details>

**Comment by mykaul** [comment]
@vponomaryov - so the question is 1st of all what causes it to shutdown while it's booting?
Also, and perhaps somewhat related - this:
```

**Comment by zimnx** [comment]
> what causes it to shutdown while it's booting?

The reason of shutdown is in my opinion inserting new CDC generation during boostrap, which results in following exception leading to startup failure:

```
INFO  2023-06-25 10:55:04,180 [shard 0] cdc - Inserting new generation data at UUID 404cf5fa-52fe-433a-9cab-5d16f2df4126
INFO  2023-06-25 10:55:04,184 [shard 0] init - Shutting down group 0 service
[...]
ERROR 2023-06-25 10:55:05,375 [shard 0] init - Startup failed: exceptions::unavailable_exception (Cannot achieve consistency level for cl ALL. Requires 3, alive 2)
```

Looks like new node is taken into CL calculations without being able to handle requests. Or the old one is still considered.

> Why are we tuning? Weren't we supposed to use pre-defined values?

When pre-computed io_properties are not mounted, iotune is executed automatically to calculate it.

> Why are we failing to tune?

Tuning is completed successfully, afaik without it Scylla would refuse to boot. These messages are just warnings about missing permiossions (Scylla container is not privileged). 

> Why does it take so long? Could be that iotune is taking 2 minutes, and overall the bootup time is too long?

In this run since first log message to the startup failure it took ~3mins, iotune takes 2mins, there's 30s gossip grace period and ~30s for the rest. That was never an issue.

**Comment by vponomaryov** [comment]
> 2. Why are we failing to tune?

Agree with @zimnx that it is not failure, Scylla would fail to start on this step in case of the tune failure.

Then, note that it is Google/GKE backend where we do not run scylla-operator -based perf tuning.

On the EKS backend we have following:
```
tuning /sys/devices/virtual/block/md127
tuning: /sys/devices/virtual/block/md127/queue/nomerges 2
warning: unable to tune /sys/devices/virtual/block/md127/queue/nomerges to 2
tuning /sys/devices/pci0000:00/0000:00:1d.0/nvme/nvme0/nvme0n1
already tuned: /sys/devices/pci0000:00/0000:00:1d.0/nvme/nvme0/nvme0n1/queue/nomerges
tuning /sys/devices/pci0000:00/0000:00:1e.0/nvme/nvme1/nvme1n1
already tuned: /sys/devices/pci0000:00/0000:00:1e.0/nvme/nvme1/nvme1n1/queue/nomerges
```

Where the md127 is non-scylla disk and the NVMEs (dedicated for scylla) are tuned ok.

**Comment by mykaul** [comment]
> The reason of shutdown is in my opinion inserting new CDC generation during boostrap, which results in following exception leading to startup failure:
> 
> ```
> INFO  2023-06-25 10:55:04,180 [shard 0] cdc - Inserting new generation data at UUID 404cf5fa-52fe-433a-9cab-5d16f2df4126
> INFO  2023-06-25 10:55:04,184 [shard 0] init - Shutting down group 0 service
> [...]
> ERROR 2023-06-25 10:55:05,375 [shard 0] init - Startup failed: exceptions::unavailable_exception (Cannot achieve consistency level for cl ALL. Requires 3, alive 2)
> ```
> 
> Looks like new node is taken into CL calculations without being able to handle requests. Or the old one is still considered.

@kbr-scylla - is this something you need to look at?

**Comment by kbr-scylla** [comment]
This error happens when you try to **bootstrap** a node to a cluster and one of the existing nodes is dead or unreachable.

In this case, from the log message it appears that there's 3 existing nodes in the cluster - and one of them is dead/unreachable.

I'm guessing that the dead node is the old incarnation of the pod - is that right? It's still a member of the cluster.
And then a new incarnation doesn't have any data of the old node - is that right? So it behaves as a completely new node. So it tries to bootstrap.

This is what our docs are saying about a similar situation - restarting an instance with ephemeral storage:
https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/cluster-management/replace-dead-node.html#setup-raid-following-a-restart
you basically need to **replace** the previous incarnation of the node.

I personally think it's crazy to run a database without actual persistent storage, and those replace operations don't guarantee that you didn't lose some data (because you did lose a node for all intents and purposes...) but oh well, this is the world we live in now.

**Comment by mykaul** [comment]
> I personally think it's crazy to run a database without actual persistent storage, and those replace operations don't guarantee that you didn't lose some data (because you did lose a node for all intents and purposes...) but oh well, this is the world we live in now.

What you call 'persistent storage' is another name for another storage product, usually... and they fail too. EBS SLAs (https://aws.amazon.com/ebs/sla/ ) aren't amazing. I've had a customer who lost twice in a single year a 20TB EBS drive. Poof. (one of the times was December 31st of that year...)

**Comment by zimnx** [comment]
> This error happens when you try to bootstrap a node to a cluster and one of the existing nodes is dead or unreachable.
> 
> In this case, from the log message it appears that there's 3 existing nodes in the cluster - and one of them is dead/unreachable.

There're 2 UN nodes, and third one is joining. 

> I'm guessing that the dead node is the old incarnation of the pod - is that right? It's still a member of the cluster.

There used to be 3 nodes in the cluster, but one was decommissioned, and new one started joining immediately after decommission is completed.

>  And then a new incarnation doesn't have any data of the old node - is that right? So it behaves as a completely new node. So it tries to bootstrap.

This is correct.

>  you basically need to replace the previous incarnation of the node.

There's no previous node, it already left the cluster using decommission procedure. The thing is new boostrap fails in the middle, and entire node get stuck. Restarting fails with:
```
ERROR 2023-06-25 10:55:12,849 [shard 0] init - Startup failed: std::runtime_error (A node with address 10.120.6.153 already exists, cancelling join. Use replace_address if you want to replace this node.)

**Comment by kbr-scylla** [comment]
> but one was decommissioned

@vponomaryov said:
> It is ~equivalent of the Scylla VM reboot.

which would mean no decommission.

Does this:
> Drain K8S node that hosts Scylla pod (kubectl drain gke-longev-scylla-operat--scylla-pool-ee517bc6-x5mp -n scylla --ignore-daemonsets --delete-local-data)

actually decommission the node?

> There's no previous node, it already left the cluster using decommission procedure. 

Let's see `nodetool status` then.

**Comment by zimnx** [comment]
Statement about simillarity to VM reboot is not accurate.

Let me explain what happens in each step without kubernetes nomenclature:

>  Drain K8S node that hosts Scylla pod (kubectl drain gke-longev-scylla-operat--scylla-pool-ee517bc6-x5mp -n scylla --ignore-daemonsets --delete-local-data)

Block VM from executing anything - it stops the Scylla node by running `"/bin/sh", "-c", "PID=$(pgrep -x scylla);supervisorctl stop scylla; while kill -0 $PID; do sleep 1; done;"` 

note: it doesn't delete any data, `--delete-local-data` parameter name is just safety check that user accepts it may happen, but in this case it doesn't.

> Wait for pod eviction

Wait until Scylla Node is stopped.

> Uncordon node

Unblock VM, previous Scylla is started again.

> Decrease number of ScyllaCluster members from 3 to 2.

Start decommission procedure - it waits until last node become UN again, and then runs `nodetool decommission`.

> Wait for the pod deletion (kubectl --namespace=scylla wait --timeout=21600s --for=delete pod sct-cluster-us-east1-b-us-east1-2)

Wait until decommission is completed. Storage backing this node is removed.

> Increase number of ScyllaCluster members from 2 to 3

Scale up the cluster by one node.

> Wait for the new pod to become up and running

Wait until new node becomes UN and opens CQL port.

---

> Let's see nodetool status then.

Scaling up is only performed when all previous nodes are "ready" - they successfully passed UN and CQL port open check. If any would fail this "readiness" precondition, new node wouldn't be added.

**Comment by kbr-scylla** [comment]
> Scaling up is only performed when all previous nodes are "ready" - they successfully passed UN and CQL port open check. If any would fail this "readiness" precondition, new node wouldn't be added.

Then we're in a pickle, because the "Cannot achieve consistency level..." error indicates the opposite.
We need to determine whether the bug is in Scylla or in Operator (e.g. maybe the readiness checks are not accurate).

I need to see `nodetool status`. Even better if I saw the logs from decommissioning as well.

Can you reproduce the problem?

**Comment by kbr-scylla** [comment]
Damn, now I see that you posted the logs. Let me look at them...

**Comment by zimnx** [comment]
> Then we're in a pickle, because the "Cannot achieve consistency level..." error indicates the opposite.
We need to determine whether the bug is in Scylla or in Operator (e.g. maybe the readiness checks are not accurate).

We check readiness periodically in 10s intervals, it could be that scaling was requested in between and it would allow it. 
But I don't see any event in the logs dump indicating any existing node failed readiness probe at all. It is generated when it happens.

**Comment by kbr-scylla** [comment]
There's a hole in the logs of the node which was supposed to decommission (`10.120.1.255`):
```
INFO  2023-06-25 10:39:12,642 [shard 0] repair - repair[24bded95-8468-4b89-9fb7-85d52bf2d4f6]: Started to shutdown off-strategy compaction updater
INFO  2023-06-25 10:39:12,643 [shard 0] repair - repair[24bded95-8468-4b89-9fb7-85d52bf2d4f6]: Finished to shutdown off-strategy compaction updater
I0625 10:52:04.829192       1 operator/sidecar.go:147] sidecar version "v1.9.0-rc.1-0-g57a2b85"
I0625 10:52:04.829277       1 flag/flags.go:64] FLAG: --burst="5"
I0625 10:52:04.829286       1 flag/flags.go:64] FLAG: --cpu-count="7"
```

according to another node's logs, decommission started at 10:50:37, which is covered by that hole...
```
INFO  2023-06-25 10:50:37,093 [shard 0] storage_service - decommission[0378fba0-8691-4a06-bf56-ad0e7fb0f23f]: Added node=10.120.1.255 as leaving node, coordinator=10.120.1.255
...
INFO  2023-06-25 10:51:07,225 [shard 0] storage_service - decommission[0378fba0-8691-4a06-bf56-ad0e7fb0f23f]: Marked ops done from coordinator=10.120.1.255
```

another suspicious thing - the decommissioning node was marked as DOWN just before decommission started. Decommission shouldn't be possible in that case:
```
INFO  2023-06-25 10:49:26,259 [shard 0] gossip - Got shutdown message from 10.120.1.255, received_generation=1687689422, local_generation=1687689422
INFO  2023-06-25 10:49:26,260 [shard 0] gossip - InetAddress 10.120.1.255 is now DOWN, status = shutdown
INFO  2023-06-25 10:50:37,093 [shard 0] storage_service - decommission[0378fba0-8691-4a06-bf56-ad0e7fb0f23f]: Added node=10.120.1.255 as leaving node, coordinator=10.120.1.255
INFO  2023-06-25 10:50:37,128 [shard 0] stream_session - [Stream #1e410480-1346-11ee-97f2-4b5af21a93e1] Executing streaming plan for Unbootstrap-audit-index-0 with peers=10.120.1.255, slave
```

and indeed the operation is suspiciously short, it doesn't look like all data was streamed, and it should have been impossible as the node was DOWN... But we do see "Marked ops done from coordinator=10.120.1.255"... what the heck happened there

**Comment by kbr-scylla** [comment]
> Wait until decommission is completed. Storage backing this node is removed.

The storage is deleted **after** decommission completes, right?

**Comment by zimnx** [comment]
> The storage is deleted after decommission completes, right?

Yes, when node reports `DECOMMISSIONED` state, we delete pod and storage.

**Comment by kbr-scylla** [comment]
Ok so here's how decommissioning procedure normally looks like at the end from the point of view of another node (I just ran it on scylla enterprise 2022.2.6):
```
INFO  2023-06-29 16:05:39,883 [shard 0] stream_session - [Stream #074ab7b0-1686-11ee-bbda-6eeb5cd1106a] Streaming plan for Unbootstrap-system_auth-index-10 succeeded, peers={127.0.0.2}, tx=0 KiB, 0.00 KiB/s, rx=0 KiB, 0.00 KiB/s                          
INFO  2023-06-29 16:05:40,194 [shard 0] gossip - Node 127.0.0.2 will be removed from gossip at [2023-07-02 16:05:39]: (expire = 1688306739899508113, now = 1688047540194772730, diff = 259199 seconds)                                                        
INFO  2023-06-29 16:05:40,195 [shard 0] storage_service - Removing tokens {5891569083958250963, -1801957846210483492, 6761589301226141281, -1972642810030720198, 5055660960045665716, -351293121356688169, -6288929715015225949, -11573406079416195, -76129678
18685268638, -4215405796995531967, -5824629450413440146, 7925106481263101325, 3164423129714089148, -3059789958325791257, -4233836524211989246, -5116689950782760799, -5696526012495579459, 4665681588695883209, -6420214376351844841, -4010263632202192184, -5
240846931412892202, -2036766769831130832, 7437834629801710383, 9193682347017883207, 8516255047077901593, -7237802272683781924, 3934599739412060715, 8592920495424640229, 3531738923751755753, 8907283758016602073, 405823573858097120, -199116954984377104, 49
06625568921063795, -5333573907166170955, 9175425182935554647, -7701927625521171330, -6202042248264103158, -8575484505450297242, 7896347011312230344, 4108320134992785103, -4163091899972093029, 7338392274101466482, 2981136863017423837, 5878107464482794251,
 -1515586811787497995, -3217409986806765764, 1207435192108993633, -4744011152323487108, 687987644155901262, 2864049838281569671, 1056360940431690887, -4616465735955870724, -7558383418825248771, -4373488553841649355, 634868176762465740, 669811290085984552
6, 6759061581522768665, -4236209220152643892, -1964856413625247689, -7972987907286020934, 1684445867642723612, 3766963229795058683, 7103841868732851735, 6882282123105331276, -6728690013046483999, 7261620148669725071, 5758379832387271424, 3575286539240471
941, 8235596116897713733, -2623946260883447626, -7612271102965767783, -3456380944548133599, -2183611716936039664, -8669196563283939302, 1581970592757522039, 112969650034705514, 2451338648497793087, 7012637638305886152, -3199413537869450022, -468415833255
1795281, 1916033276441130367, 6946980288019400021, 4969785236593030227, -5995538200960899882, 2787899180153472801, 8596299046217523245, 6710357327815976879, 4598447466266687333, -3896117424651867426, 642231597183680851, 2973851061852598764, -321508553661
1305022, 2636387050986286096, 8558675338646266372, 4022912620326448551, -5877028510862360511, 5891608817011632369, -7615351374535714105, 9159316343946517681, 3990181558742546374, -6750771856108336766, -9222633084096348542, 4330539292109454160, 2131505586
286696467, -4952478532588459199, 2556879899042442167, -4741134042886396398, -574297207068965962, -4584370912183712015, -2063643818194800679, 6002591633986977440, 5786508888547610666, -2588397006537416406, -2971214030080898645, -4262033170801603016, -2868
40196699351118, 1990907309974865103, -5591358304622260170, -8988146124481117227, 421539849764308880, -5143184340335694676, -7442999523011258416, -1292828821037132190, -4393183424379852738, -2959220607272731644, -8064463598950266147, -8226433608644107256,
 1855569260860477504, -4959757792466921240, 2744634592339459295, 1474778625187065066, 4770212161102269675, -4402319866272674998, 2710034273346149680, -7986095663784299147, 2157205338394439983, 2071612741401544820, -6998935395699385993, 575520236709430596
9, -6462328348455560088, 7841524301179407002, -1399328344266472519, -1120313997850781978, -8599702780747966501, -210847929614665716, 7514186128614689978, -8085701938194137344, 2706950342435431446, 211507491429098115, -7892852753194670791, -15591030952446
21719, 4723660201069623433, -1225163413944459872, 5717233751622776413, 2342441655762572304, -6427359707364538731, 637418000688001085, 6970904598422542663, 5146415439452725897, -25422915676712385, 6649866326744631618, -5211892908658481686, -43690578029892
44970, 2615555227135683246, 6413403091182401257, -5057304478649686256, 5746140791362276195, 3827712540810909402, 4205383060190970556, 1267241022423273367, 4160163310654411900, 1488591453329140369, -7832718505636080155, -3711063816622276345, -284983708971
9865093, 1288223407530636047, 3572324792515544401, -6619901761144201324, 7324415076274718559, 6084484480691701218, -7585815297553489763, 8269240340771774635, -8928027789900900127, 6857166315025783420, -7385411659082477419, 8023193130664801367, 7085168173
315125230, 2552127622274697363, 7761815733020575750, -3223862428945766045, 8639231535971466213, 3865123480479784394, 3499268377595324265, -4426015757790117489, -7778978999026834949, -5390172254426170006, 4600115182906081174, -8892456220877644978, -211670
2421130397275, 3025862109772137024, 4317804159152278242, 4611977857296684530, 5115063156912268223, -2804137320144516000, 5773098484810457551, 185702204706576014, 1236790267015536870, 345622094502841219, 6227313664226761097, -5735561059512132415, -3289982
752412429893, 2768486774644870220, 5273369657535599581, -533940186690968065, 803783346804276156, 7354110770523129838, -8952625333106090069, -7977506939734547696, -5048982680316240893, -3708803705540387483, -2642663464660368717, -2960643403632944482, 2510
205024673851211, 1395394505691690766, -524577220007845200, -7067133790929521993, 496879461396270822, -2666133930907725733, 4418846256660093981, 4761150074265132087, -2029801115886355352, 3679203360312358301, -6376084871992832902, -947506311437654344, 506
2578333885256850, -479951939399235969, 4988812404142835979, 4884270679766597084, -6477465439194546424, 782492741694363039, -610142189407412518, -3965854654281710472, -6882992567794959263, -3588996072610472241, 2822416030099175212, 2616678417117679436, -1
390732295092675885, 5748702660617835611, -6896547504755117443, -2370843266237669662, 1632931439940477479, -1430064313504389658, -4183406300768515541, -6157263846578177315, -5041046899484256798, -3256155318832188590} for 127.0.0.2                         
INFO  2023-06-29 16:05:40,260 [shard 1] compaction - [Compact system.peers 07843e40-1686-11ee-a399-54a02fa2bbf1] Compacting [workdir/data/system/peers-37f71aca7dc2383ba70672528af04d4f/me-13-big-Data.db:level=0:origin=memtable,workdir/data/system/peers-37
f71aca7dc2383ba70672528af04d4f/me-10-big-Data.db:level=0:origin=compaction]                                                                                                                                                                                   
INFO  2023-06-29 16:05:40,279 [shard 1] compaction - [Compact system.peers 07843e40-1686-11ee-a399-54a02fa2bbf1] Compacted 2 sstables to [workdir/data/system/peers-37f71aca7dc2383ba70672528af04d4f/me-16-big-Data.db:level=0]. 86kB to 40kB (~47% of origina
l) in 15ms = 2MB/s. ~256 total partitions merged to 1.                                                                                                                                                                                                        
INFO  2023-06-29 16:05:41,465 [shard 0] gossip - InetAddress 127.0.0.2 is now DOWN, status = LEFT                                                                                                                                                             
INFO  2023-06-29 16:05:42,922 [shard 0] storage_service - decommission[0c8275ab-8db5-487c-b874-eed822407146]: Marked ops done from coordinator=127.0.0.2                                                                                                      

... some time passes, compactions run ...

INFO  2023-06-29 16:06:40,468 [shard 0] gossip - 60000 ms elapsed, 127.0.0.2 gossip quarantine over                                                                                                                                                           
```

important stuff:
```
INFO  2023-06-29 16:05:40,194 [shard 0] gossip - Node 127.0.0.2 will be removed from gossip at [2023-07-02 16:05:39]: (expire = 1688306739899508113, now = 1688047540194772730, diff = 259199 seconds)                                                        
```
and
```
INFO  2023-06-29 16:05:41,465 [shard 0] gossip - InetAddress 127.0.0.2 is now DOWN, status = LEFT                                                                                                                                                             
```

these things are **missing** in the logs you provided.

It looks like other nodes never got the gossip LEFT event from the decommissioning node... so they still considered it a member.

I have two hypotheses why this could happen:
- you shutdown the node too early; it didn't manage to actually gossip with the other nodes that it has LEFT before it got shutdown
- OR you actually started shutting it down concurrently, even before decommissioning has started

This second hypothesis because of this DOWN event that we saw in the logs from the decommissioning node *before* decommission ran:
```
INFO  2023-06-25 10:49:26,260 [shard 0] gossip - InetAddress 10.120.1.255 is now DOWN, status = shutdown
INFO  2023-06-25 10:50:37,093 [shard 0] storage_service - decommission[0378fba0-8691-4a06-bf56-ad0e7fb0f23f]: Added node=10.120.1.255 as leaving node, coordinator=10.120.1.255
```

another possibility is that this DOWN event comes from this "pod eviction" event.
However what's suspicious is that we never get an UP event after it.
If the node restarted before decommission, it should have been observed as UP again before we saw the decommissioning logs.

Unfortunately, due to the hole in the logs of `sct-cluster-us-east1-b-us-east1-1` I cannot determine what happened exactly.

Anyway to avoid flakiness, we need to actually confirm that other nodes saw the LEFT status before we can safely shutdown the node.

I know, it's bad... but that's the gossiping world. (Which we're trying to escape.)

**Comment by kbr-scylla** [comment]
> Anyway to avoid flakiness, we need to actually confirm that other nodes saw the LEFT status before we can safely shutdown the node.

> I know, it's bad... but that's the gossiping world. (Which we're trying to escape.)

BTW. now I found that this is fixed in new versions
```
commit 693d71984f33763816f05652b22ed2c2bd8fddb5
Author: Asias He <asias@scylladb.com>
Date:   Wed Feb 1 10:13:48 2023 +0800

    storage_service: Check node has left in node_ops_cmd::decommission_done
    
    In test with ring delay zero, it is possible that when the
    node_ops_cmd::decommission_done is received, the nodes remained in the
    cluster haven't learned the LEFT status for the leaving node yet.
    
    To guarantee when the decommission restful api returns, all the nodes
    participated the decommission operation have learned the LEFT status, a
    check in the node_ops_cmd::decommission_done is added in this patch.
    
    After this patch, the decommission tests which start multiple
    decommission in a loop with ring delay zero in
    test/topology/test_topology.py passes.
```

so the decommissioning node will not consider the operation to finish before other nodes get the LEFT status.

But 2022.2.6 doesn't have this patch.

**Comment by kbr-scylla** [comment]
But the decommissioning node **should** have slept for ring_delay before shutting down
```cpp
future<> storage_service::leave_ring() {
    co_await _sys_ks.local().set_bootstrap_state(db::system_keyspace::bootstrap_state::NEEDS_BOOTSTRAP);
    co_await mutate_token_metadata([this] (mutable_token_metadata_ptr tmptr) {
        auto endpoint = get_broadcast_address();
        tmptr->remove_endpoint(endpoint);
        return update_topology_change_info(std::move(tmptr), ::format("leave_ring {}", endpoint));
    });

    auto expire_time = _gossiper.compute_expire_time().time_since_epoch().count();
    co_await _gossiper.add_local_application_state(gms::application_state::STATUS,
            versioned_value::left(co_await _sys_ks.local().get_local_tokens(), expire_time));
    auto delay = std::max(get_ring_delay(), gms::gossiper::INTERVAL);
    slogger.info("Announcing that I have left the ring for {}ms", delay.count());
    co_await sleep_abortable(delay, _abort_source);
}
```

so in theory there was plenty of time for other nodes to observe the LEFT status.

The fact that they didn't indicates some problem. Maybe it was just network problem.
I recall you already encountered problems with gossiper messages not reaching other nodes in K8s environment?

Again, hard to say what exactly happened because we don't have the logs of the decommissioning node from that time for some reason :/

**Comment by zimnx** [comment]
We saw it also on 2023.1.0-rc5, and there's no log hole during decommission: https://github.com/scylladb/scylla-operator/issues/1241#issuecomment-1584786757

Looks like this patch isn't available in any official version:
```
$ git tag --contains=693d71984f33763816f05652b22ed2c2bd8fddb5
scylla-5.3.0-rc0
```

**Comment by rzetelskik** [comment]
> I recall you already encountered problems with gossiper messages not reaching other nodes in K8s environment?

You're probably thinking of https://github.com/scylladb/scylla-operator/issues/982 and https://github.com/scylladb/scylladb/issues/11302.

Looks like we might have hit what Kostja was describing here https://github.com/scylladb/scylladb/issues/12892?

**Comment by kbr-scylla** [comment]
> Looks like we might have hit what Kostja was describing here https://github.com/scylladb/scylladb/issues/12892?

What do you mean? The scenario and symptoms are different.

**Comment by rzetelskik** [comment]
> What do you mean? The scenario and symptoms are different.

I wasn't referring to the original issue but the comments after https://github.com/scylladb/scylladb/issues/12892#issuecomment-1433152085 and your earlier comment here
> It looks like other nodes never got the gossip LEFT event from the decommissioning node... so they still considered it a member.

But it looks like 2022.2.6 doesn't have `58c65954b8cedc498678e97027885aea1cad24f3`, which supposedly fixed https://github.com/scylladb/scylladb/issues/11302, either.

**Comment by kbr-scylla** [comment]
Thanks @rzetelskik .

Based on the logs and my earlier analysis --
this looks like an exact duplicate of https://github.com/scylladb/scylladb/issues/11302. The scenario of https://github.com/scylladb/scylladb/issues/11302 happens, the node "decommissions" but really the rest of the cluster still considers it as a peer. And then we try to bootstrap a new node.

As @rzetelskik observed, the version on which OP ran the test, `2022.2.6`, does not have the fix for #11302.

Closing as duplicate.
